### DISCUSSION TITLE
Vite config exposed for Cypress?

### DISCUSSION THREAD

**Discussion by Zielgestalt** [nuxt/nuxt#16515] [Questions] [answered]
I‘m trying to setup Cypress for nuxt3 component testing with vite as testserver. Ideally I can use the same vite config for cypress that nuxt is using. 

Is the vite config exposed somewhere so I can use it with cypress?

Thanks!

**Comment by thegouge** [comment]
I was wondering the same thing, actually. 
Also, @Zielgestalt did you manage to get cypress component testing working in your Nuxt 3 project?

**Comment by madsh93** [comment]
Did any of you find a way to install Cypress with Nuxt 3?

**Comment by Zielgestalt** [comment]
@madsh93  I found this repo but I didn't have time to try it out. But maybe that helps? https://github.com/mikespinks0401/Nuxt3-TailwindCss-CypressE2E-UnitTesting-Boilerplate

**Comment by madsh93** [comment, marked as answer]
@pautomagi came up with a solution that works:

It seems Cypress official guide just passes an empty vite config with the `@vitejs/plugin-vue` as a plugin. This solution passes this.

```js
const { startDevServer } = require('@cypress/vite-dev-server');
const vue = require('@vitejs/plugin-vue');

module.exports = (on, config) => {
    on('dev-server:start', (options) => {
        return startDevServer({
            options,
            viteConfig: {
                plugins: [vue()],
            },
        });
    });

    return config;
};
```

**Reply by Zielgestalt** [reply]
Thanks!

**Reply by jclaessens97** [reply]
Not entirely sure where I should configure this? Can you give any more pointers?

**Reply by joesobo** [reply]
@madsh93 seconding this request. Can you give some more info about how you got it to work? With cypress 10 plugins are no longer available as well as the `on('dev-server:start') event`
